core/src/main/scala/replpp/Operators.scala

```diff
@@ -135,15 +135,23 @@ object Operators {
         val chunkSize = 8192
         var remaining = bytes.length
         var pos = 0
-        while (remaining > 0) {
-          val currentWindow = math.min(remaining, chunkSize)
-          stdin.buffered.write(value, pos, currentWindow)
-          pos += currentWindow
-          remaining -= currentWindow
+        var stopped = false
+        Using.resource(stdin) { stdin =>
+          while (remaining > 0 && !stopped) {
+            val currentWindow = math.min(remaining, chunkSize)
+            try {
+              stdin.buffered.write(value, pos, currentWindow)
+              pos += currentWindow
+              remaining -= currentWindow
+            } catch {
+              case t: Throwable =>
+                // most likely the user exited the subprocess
+                stopped = true
+            }
+          }
+          // flush stdin, but ignore errors
+          try {stdin.flush()} catch { case t: Throwable => /*ignore*/ }
```

Comment:
should be safe to just drop this line - closing already implies pushing out all unwritten data.

Reply:
that's what I thought too, but unfortunately that's not the case. 
check out the two runs above, I only added this line after seeing the red tests, which were caused by non-flushiness